graph/gnuplot.h

```diff
@@ -0,0 +1,188 @@
+  explicit WithMeta(std::function<void(Plotter&)> f) : f_(f) {}
+  WithMeta& Name(const std::string& name) {
+    meta_ += " t " + Escape(name);
+    return *this;
+  }
+  // TODO(dkorolev): Add more setters here.
+};
+
+struct GNUPlot {
+  std::string output_format_ = "png";
+  size_t x_ = 800;
+  size_t y_ = 800;
+  std::map<std::string, std::string> parameters_;
+  std::vector<std::unique_ptr<PlotDataBase>> plots_;
+
+  GNUPlot() = default;
+
+  GNUPlot& TermSize(size_t x, size_t y) {
+    x_ = x;
+    y_ = y;
+    return *this;
+  }
+
+  GNUPlot& Title(const std::string& title) {
+    parameters_["title"] = Escape(title);
+    return *this;
+  }
+
+  GNUPlot& Grid(const std::string& title) {
+    parameters_["grid"] = Escape(title);
+    return *this;
+  }
+
+  GNUPlot& NoKey() {
+    parameters_["key"] = "";
+    return *this;
+  }
+
+  GNUPlot& KeyTitle(const std::string& key_title) {
+    parameters_["key"] = "title " + Escape(key_title);
+    return *this;
+  }
+
+  GNUPlot& XRange(double x, double y) {
+    parameters_["xrange"] = strings::Printf("[%+lf:%+lf]", x, y);
+    return *this;
+  }
+
+  GNUPlot& YRange(double x, double y) {
+    parameters_["yrange"] = strings::Printf("[%+lf:%+lf]", x, y);
+    return *this;
+  }
+
+  GNUPlot& OutputFormat(const std::string& output_format) {
+    output_format_ = output_format;
+    return *this;
+  }
+
+  // TODO(dkorolev): `template<typename T> GNUPlot& Plot(T&& plot)` to support pre-populated plots as well.
+  GNUPlot& Plot(std::function<void(Plotter&)> plot) {
+    plots_.emplace_back(new PlotDataFromFunction(plot));
+    return *this;
+  }
+
+  GNUPlot& Plot(const WithMeta& plot) {
+    plots_.emplace_back(new PlotDataFromFunction(plot.f_, plot.meta_));
+    return *this;
+  }
+
+  operator std::string() const {
+    assert(!plots_.empty());  // TODO(dkorolev): Exception?
+
+    const std::string input_file_name = bricks::FileSystem::GenTmpFileName();
+    const std::string output_file_name = bricks::FileSystem::GenTmpFileName();
+    const auto input_deleter = bricks::FileSystem::ScopedRemoveFile(input_file_name);
+    const auto output_file_deleter = bricks::FileSystem::ScopedRemoveFile(output_file_name);
+    {
+      bricks::FileSystem::OutputFile f(input_file_name);
+      f << "set term " << output_format_ << " size " << x_ << ',' << y_ << std::endl;
+      for (const auto& p : parameters_) {
+        if (!p.second.empty()) {
+          f << "set " << p.first << ' ' << p.second << std::endl;
+        } else {
+          f << "unset " << p.first << std::endl;
+        }
+      }
+      for (size_t i = 0; i < plots_.size(); ++i) {
+        f << (i ? ", " : "plot ") << "'-'";
+        plots_[i]->AppendMetadata(f);
+      }
+      f << std::endl;
+      for (const auto& p : plots_) {
+        p->AppendData(f);
+        f << "end" << std::endl;
+      }
+    }
+    ::system(strings::Printf("gnuplot <%s >%s\n", input_file_name.c_str(), output_file_name.c_str()).c_str());
```

Comment:
I'll make a platform-neutral ```::system()``` call and throw an exception -- when merging a few places I use ```::system()``` from as of today.